### ISSUE TITLE
Move 'cmake_test/' into 'cmake/'

### ISSUE THREAD

**Issue by zachcran** [CMakePP/CMakeTest#74]
**Is your feature request related to a problem? Please describe.**
CMakeTest has a layout that differs from the rest of CMakePP org repositories. They have CMake source files contained under the `cmake/` directory.

**Describe the solution you'd like**
The `cmake_test` directory should be moved inside the `cmake` directory.

**Comment by zachcran** [comment]
With this proposed move, I think everything in `tests/` should also be moved into `tests/cmake_test`. This then might allow `test_directory_2` to be moved into `tests/`, greatly cleaning up the top-level directory.

**Comment by AutonomicPerfectionist** [comment]
I've meant to do that for a while; I've just been tied up with other things. I can start shuffling things around next week. Before merging such a drastic change I would like to do some more extensive testing to make sure the module path or any include references aren't borked, we can do that with CMaize/CMakePPLang right? Just changing the repo they pull CMakeTest from?

**Comment by zachcran** [comment]
Yes, that should work to do more extensive testing. The integration tests I have been working on for CMaize ([here](https://github.com/CMakePP/CMaize/pull/99)) are a bunch of other projects that use CMakeTest as well (although I haven't quite gotten the CI for them working yet), so we have tons of testing we can do.
